core-processor/src/processing.rs

```diff
@@ -207,28 +230,37 @@ pub fn process_error(
         journal.push(JournalNote::SystemReserveGas { message_id, amount });
     }
 
-    if system_reservation_ctx.has_any() {
-        if !dispatch.is_error_reply()
+    if let ProcessErrorCase::ExecutionFailed {
+        reason,
+        allow_send_signal,
+    } = &case
+    {
+        // TODO: consider to handle error reply and init #3701
+        if *allow_send_signal
```

Comment:
why do you need this allowance if you keep checking kinds right in place

Reply:
because we does not send signal in case of ExecutionError::Actor, but still need to handle this this error as execution. I is also possible to make additional variant in enum if you want

Reply:
ExecutionError::Actor doesnt anything (and wasn't, I double checked after your answer) due to empty reservation context (its default used), so this field is unnecessary I guess ?

Reply:
You can have not empty context. If your message was sent before runtime upgrade, which by some reason now will rejects your program on instantiation, then you need to unreserve system reservation, but no need to send signal.

Reply:
new issue created